code/game/gamemodes/wizard/spellbook.dm

```diff
@@ -259,6 +267,9 @@
 			user << "\blue The walls suddenly disappear."
 	return 1
 
+/datum/spellbook_entry/item/bloodbottle/CanBuy(var/mob/living/carbon/human/user,var/obj/item/weapon/spellbook/book)
+	return ..() && !limit

```

Comment:
A proc redefine shouldn't be needed here, considering when limit is 0, spellbook_entry/CanBUy will return 0.